### PR TITLE
Added m47ch4n keymap

### PR DESCRIPTION
**Pull request by m47ch4n**
## Description
Added m47ch4n keymap for ergodash/rev1

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c [hunk 1]
```diff
@@ -0,0 +1,226 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c]: This line is not actually needed, and should be removed. 

```suggestion
```

File: keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+#define CASE_US(CODE, US, JP)                   \
+    case CODE:                                  \
+        (LAYOUT_STATUS == JP_LAYOUT ? JP : US); \
+        return false;
+
+bool pushed_lower_l = false;
+bool pushed_lower_r = false;
+bool pushed_adjust  = false;
+
+void update_layer(void) {
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c]: I think you may want to do this, instead: 
https://docs.qmk.fm/#/ref_functions?id=update_tri_layer_statestate-x-y-z

I think you're trying to do the same thing, and this is a much simpler way to do so.

File: keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c (hunk 1 above)
Comment (m47ch4n) [reply keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c]: I appreciate your code review!
I have fixed code according your suggestions https://github.com/qmk/qmk_firmware/pull/6673/commits/ba9fc0ee6c5b34f922e018f8860296ac288d467a.
Thank you.

File: keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c (hunk 2 above)
Comment (m47ch4n) [reply keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c]: I fixed my code using `update_tri_layer` instead.
You can see my changes here https://github.com/qmk/qmk_firmware/pull/6673/commits/ba9fc0ee6c5b34f922e018f8860296ac288d467a.

**Pull request merged**